### PR TITLE
feat: breaking changes in slider component

### PR DESCRIPTION
**Pull request by 0xgabii**

### CODE REVIEW

**Review by graykode — CHANGES_REQUESTED** [review]
.

**Review by 0xgabii — COMMENTED** [review]
.

File: pages/deposit/[symbol].tsx [hunk 1]
```diff
@@ -101,14 +100,13 @@ const Deposit = () => {
...
-                  disabled:
-                    amount === 0n || epochs === 0 || amount > maxDepositAmount,
+                  disabled: amount === 0n || amount > maxDepositAmount,
                   onClick: async () => {
                     const { timestamp } = await publicClient.getBlock()
                     const hash = await deposit(
                       asset,
                       amount,
-                      epochs,
+                      epochs + 1,
```
Comment (graykode) [inline pages/deposit/[symbol].tsx:109]:
```suggestion
                      epochs,
```
@Dorvin Let's move +1 logic to deposit context

Comment (graykode) [inline pages/deposit/[symbol].tsx:109]: @Dorvin Include a TODO comment for when the contract interface is changed.

File: containers/form/borrow-form-container.tsx [hunk 2]
```diff
@@ -235,7 +231,7 @@ const BorrowFormContainer = ({
             collateralAmount,
             asset,
             borrowAmount,
-            epochs,
+            epochs + 1,
```
Comment (graykode) [inline containers/form/borrow-form-container.tsx:234]: p2; same with deposit @Dorvin

File: containers/modal/edit-expiry-modal-container.tsx [hunk 3]
```diff
@@ -37,31 +37,29 @@ const EditExpiryModalContainer = ({
...
-          .slice(expiryEpochIndex, epochs)
+          .slice(expiryEpochIndex + 1, epochs + 1)
           .reduce((acc, { interest }) => acc + interest, 0n),
         data
-          .slice(expiryEpochIndex, epochs)
+          .slice(expiryEpochIndex + 1, epochs + 1)
           .reduce((acc, { payable }) => acc && payable, true),
         data
-          .slice(epochs - 1, expiryEpochIndex - 1)
+          .slice(epochs + 1, expiryEpochIndex + 1)
```
Comment (graykode) [inline containers/modal/edit-expiry-modal-container.tsx:50]: p1; @Dorvin 
```suggestion
          .slice(epochs, expiryEpochIndex + 1)
```

File: containers/modal/edit-expiry-modal-container.tsx [hunk 4]
```diff
@@ -37,31 +37,29 @@ const EditExpiryModalContainer = ({
...
           .reduce((acc, { refund }) => acc + refund, 0n),
         data
-          .slice(epochs - 1, expiryEpochIndex - 1)
+          .slice(epochs + 1, expiryEpochIndex + 1)
```
Comment (graykode) [inline containers/modal/edit-expiry-modal-container.tsx:53]: p1; @Dorvin 
```suggestion
          .slice(epochs, expiryEpochIndex + 1)
```

**Comment by graykode** [comment]
@Dorvin I found bug when calculate expected refund amount: https://github.com/clober-dex/coupon.finance/pull/303/commits/5b75db9fc3e50f81ff47e5385ee9ce407f5796e9

**Review by Dorvin — APPROVED** [review]

**Pull request merged**